interfaces/global-state-types.ts

```diff
@@ -10,4 +10,5 @@ export interface GlobalStateInterface {
   providerToken?: string | null;
   userId?: number | null;
   hasReports?: boolean;
+  openSearch?: boolean;
```

Comment:
Curious why this is global?

Reply:
Great question haha 😅

It was getting somewhat uneasy to manage the `openSearch` state, as I have started to drill down the `setOpenSearch` dispatch function almost 4 components deep to the search results list item.

State Tree

- AuthSection (Declares the `openSearch` state)
  - SearchDialogTrigger (uses the `setOpenSearch`)
  - SearchDialog (receives the `setOpenSearch` as prop from AuthSection to close the dialog onClick/keyboard shortcuts)
    - SearchResult (receives the `setOpenSearch` as prop from SearchDialog to pass it down to its child component`
      - UserResultCard - (receives the `setOpenSearch` as prop from `SearchResult` to close dialog onClick/keyboardCtrl result item.

Reply:
Nice! 👍. Looks good to me